src/url.cpp

```diff
@@ -469,6 +474,13 @@ namespace ada {
     if(checkers::is_ipv4(host.value())) {
       return parse_ipv4(host.value());
     }
+
+    // The domain label size ehckers should only be applied for fully qualified domain names.
+    // FQDNs have label hierarchy (meaning at least one dot).
+    if(host.value().find_first_of(".") != std::string_view::npos) {
```

Comment:
Why not do `checkers::check_domain(host.value());` directly? The if clause is almost always going to be true anyhow, so it saves us nothing, and you are inlining check_domain. In check_domain, you could easily acomodate the case where there is no dot: just unroll your loop once: process the first search for a '.' separately. In this way, you will avoid doing the work twice.

Reply:
I agree with @lemire: We can just return checkers::check_ domain and remove the branch.